modules/@apostrophecms/doc-type/index.js

```diff
@@ -715,15 +723,47 @@ module.exports = {
         await self.emit('afterRevertDraftToPublished', req, result);
         return result.draft;
       },
+      // Used to implement "Undo Publish."
+      //
+      // Revert the doc `published` to its content as of its most recent
+      // previous publication. If it has only been published once ever,
+      // delete the published version of the document completely.
+      // Does not affect the related draft document.
+      //
+      // Returns `false` if it is not possible to revert because
+      // only one previous version is available and it has already
+      // been used in this way.
       async revertPublishedToPrevious(req, published) {
+        if (published.publicationCount === 1) {
```

Comment:
Couldn't we look for the `:previous` doc and delete if that doesn't exist? I don't like the idea of adding a `publicationCount` property onto all Apos documents for this one method when there's another option.

Reply:
I considered that but I feel it is too risky, as if I'm a third party developer looking to prune the db in various ways I would not assume that deleting the "previous published" could lead to total deletion of published documents any scenario.